statement.go

```diff
@@ -46,14 +46,37 @@ func (s *stmt) NumInput() int {
 }
 
 func (s *stmt) bind(args []driver.NamedValue) error {
-	if s.NumInput() != len(args) {
+	if s.NumInput() > len(args) {
 		return fmt.Errorf("incorrect argument count for command: have %d want %d", len(args), s.NumInput())
 	}
 
 	// FIXME (feature): we can't pass nested types as parameters (bind_value) yet
 
-	for i, v := range args {
-		switch v := v.Value.(type) {
+	// relaxed length check allow for unused parameters.
+	for i := 0; i < s.NumInput(); i++ {
+		name := C.duckdb_parameter_name(*s.stmt, C.idx_t(i+1))
+		paramName := C.GoString(name)
+		C.duckdb_free(unsafe.Pointer(name))
+		//t := C.duckdb_param_type(*s.stmt, C.idx_t(i+1))
+
+		// fallback on index position
+		var arg = args[i]
+
+		// override with ordinal if set
+		for _, v := range args {
+			if v.Ordinal == i+1 {
+				arg = v
+			}
+		}
+
+		// override with name if set
+		for _, v := range args {
+			if v.Name == paramName {
+				arg = v
+			}
+		}
```

Comment:
Can we merge these two loops?

Reply:
It is in 3 passes to set different priorities. For example in case the client code or a wrapper library is setting/generating `NamedValues` with both a `Name` and an `Ordinal` as a fallback mechanism.